### DISCUSSION TITLE
Inconsistent indeterminate checkbox behavior

### DISCUSSION THREAD

**Discussion by Omikorin** [chakra-ui/ark#3182] [General]
### Description

When using `indeterminate` prop in the Checkbox it behaves differently across frameworks.

In React:
- With `indeterminate` on, it is impossible to toggle state by clicking

In Vue, Svelte:
- Clicking on the indeterminated checkbox changes state to `checked = true`

Solid: I couldn't check because of some broken build

Zag.js:
- Machine seems to be able to update state from `indeterminate` to boolean values.

In terms of design, I found that many software installers have the Vue/Svelte option - possibility to remove the intermediate state. 

### Link to Reproduction (or Detailed Explanation)

Storybook

### Steps to Reproduce

1. Use Storybook
2. You may check on [Zag.js website ](https://zagjs.com/components/react/checkbox) too
3. Set `indeterminate` prop on the checkbox's root
4. Click on the checkbox

### Ark UI Version

4.5.0

### Framework

- [X] React
- [X] Solid
- [X] Vue

### Browser

Brave 1.73.104

**Comment by segunadebayo** [comment]
The `indeterminate` is not a prop per se. The correct prop is `checked=indeterminate`

See the Zag.js website implementation here:
https://github.com/chakra-ui/zag/blob/8d2f0bc635c7be9bf98affda3eb8a191ea35c93b/website/components/machines/checkbox.tsx#L11

Vue uses two way binding for the checked state and we need to decide on the expected behavior.

**Comment by segunadebayo** [comment]
Closing — `Checkbox.Indicator` properly supports the `indeterminate` state. This has been fixed.
